map/bookmark_manager.cpp

```diff
@@ -96,6 +103,115 @@ BookmarkManager::SharingResult GetFileForSharing(BookmarkManager::KMLDataCollect
   return {{categoryId}, std::move(tmpFilePath)};
 }
 
+bool addFileToArchive(std::string const & filePath, std::string const & fileInArchive, zip::File zip)
+{
+  zip::FileInfo zipInfo = {};
+  CreateTMZip(zipInfo.tmz_date);
+  if (zip::Code::Ok != zip::OpenNewFileInZip(zip, fileInArchive, zipInfo, "",Z_DEFLATED, Z_DEFAULT_COMPRESSION))
+    return false;
+
+  try
+  {
+    base::FileData file(filePath, base::FileData::OP_READ);
+    uint64_t const fileSize = file.Size();
+
+    uint64_t currSize = 0;
+    std::array<char, zip::kFileBufferSize> buffer;
+    while (currSize < fileSize)
+    {
+      auto const toRead = std::min(buffer.size(), static_cast<size_t>(fileSize - currSize));
+      file.Read(currSize, buffer.data(), toRead);
+
+      if (zip::Code::Ok != zip::WriteInFileInZip(zip, buffer, toRead))
+        return false;
+
+      currSize += toRead;
+    }
+  }
+  catch (Reader::Exception const & ex)
+  {
+    LOG(LERROR, ("Error reading file:", filePath, ex.Msg()));
+    return false;
+  }
+  return true;
+}
+
+
+std::string buildIndexFile(std::vector<std::string> filesForIndex)
```

Comment:
Don't pass containers by value if a copy is not needed.
```suggestion
std::string buildIndexFile(std::vector<std::string> const & filesForIndex)
```

Reply:
Fixed